addons/sale/models/sale_order.py

```diff
@@ -507,6 +511,36 @@ def _compute_amounts(self):
             order.amount_tax = tax_totals['tax_amount_currency']
             order.amount_total = tax_totals['total_amount_currency']
 
+    def _pay_term_is_early_discount_mixed(self):
+        self.ensure_one()
+        pay_term = self.payment_term_id
+        return pay_term and pay_term.early_discount and pay_term.early_pay_discount_computation == 'mixed'
+
+    def _add_lines_for_early_payment_discount_tax_correction(self):
+        self.ensure_one()
+        epd_lines = []
+        percentage = self.payment_term_id.discount_percentage
```

Comment:
```suggestion
        percentage = self.payment_term_id.discount_percentage
```
From what i see on the account_move model, it can be falsy (0 i suppose), shouldn't we return early in this case too?